src/theory/quantifiers/sygus/rcons_obligation.h

```diff
@@ -25,37 +25,46 @@ namespace theory {
 namespace quantifiers {
 
 /**
- * A utility class for Sygus Reconstruct obligations. An obligation is a builtin
+ * A class for holding Sygus Reconstruct obligations. An obligation is a builtin
  * term t and a SyGuS type T, and indicates that we are trying to build a term
- * of type T whose builtin analog is equivalent to t. The main algorithm encodes
- * each obligation (t, T) as a skolem k of type T to embed obligations in
- * candidate solutions (see d_candSols below). It mainly operates over skolems
- * and stores cooresponding builtin terms and other info in instances of this
- * class. Notice that the SyGuS type T of an obligation is not stored in this
- * class as it can be inferred from the type of the skolem k.
+ * of type T whose builtin analog is equivalent to t. This class encodes each
+ * obligation (T, t) as a skolem k of type T to embed obligations in candidate
+ * solutions (see d_candSols below). Notice that the SyGuS type T of an
+ * obligation is not stored in this class as it can be inferred from the type of
+ * the skolem k.
  */
-class RConsObligationInfo
+class Obligation
```

Comment:
I think I'd prefer the name of this class was still specific to reconstruction, maybe RConsObligation, esp. since we don't enclose this in a namespace.